modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -146,8 +146,11 @@ default InternalTransaction beginExplicitRo(HybridTimestampTracker timestampTrac
      * @param txId Transaction id.
      * @param ts The timestamp which is associated to txn completion.
      * @param commit {@code True} if a commit requested.
+     * @param timeoutExceeded {@code True} if a timeout exceeded. MUST: 'commit' != 'timeoutExceeded'.
```

Comment:
`'commit' != 'timeoutExceeded'` - but can they both be `false`?

Reply:
yes, fixed